lib/GenomeFileUtil/core/GenomeUtils.py

```diff
@@ -229,3 +229,152 @@ def check_full_contig_length_or_multi_strand_feature(feature, is_transpliced, co
     if len(strand_set) > 1 and not is_transpliced:
         feature["warnings"] = feature.get('warnings', []) + [warnings["both_strand_coordinates"]]
     return feature 
+
+def check_feature_ids_uniqueness(genome):
+    """
+    Tests that all feature ids in a genome are unique across all 4 feature type lists
+    Returns dict of Non Unique IDS and the counts associated with them. 
+    If all IDS are unique, it then returns an empty dict
+    """
+    unique_feature_ids = set()
+    duplicate_feature_id_counts = dict()
+    feature_lists = ["features","cdss","mrnas","non_coding_features"]
+
+    for feature_list in feature_lists:
+        for feature in genome[feature_list]:
+            if feature["id"] in unique_feature_ids:
+                if feature["id"] in duplicate_feature_id_counts:
+                    duplicate_feature_id_counts[feature["id"]] += 1
+                else:
+                    duplicate_feature_id_counts[feature["id"]] = 2
+            else:
+                unique_feature_ids.add(feature["id"])
+    if len(unique_feature_ids) == 0:
+        duplicate_feature_id_counts["WARNING NO FEATURE IDS FOUND IN GENOME."] = 0
+    return duplicate_feature_id_counts
+        
+def make_id_set(feature_list):
+    '''
+    Helper function to make id lookup sets for a feature list
+    '''
+    return set((x['id'] for x in feature_list))
+
+def confirm_feature_relationships(genome, feature_id, feature_id_sets_dict = dict()):
+    """
+    Finds that feature id in the genome and verifies if all feature ids in relationships are present
+    Note it does not check if a relationship field is present as some features will not have determined relationships.
+    returns dict of types as keys and ids that were not found. An empty dict means all declared relationships were found
+    """
+    not_found_relationships = dict()
+    if len(feature_id_sets_dict) == 0:
+        # then need to initialize the lists. Only needed if not called from confirm_genomes_feature_relationships
+        feature_lists = ["features","cdss","mrnas","non_coding_features"]
+        for feature_list in feature_lists:
+            if feature_list in genome:
+                feature_id_sets_dict[feature_list] = make_id_set(genome[feature_list])
+            else:
+                feature_id_sets_dict[feature_list] = set()
```

Comment:
This works but could be a lot more succinct with dictionary "get" method: https://docs.python.org/3/library/stdtypes.html#dict.get

This allow you to specify a type to return if the key is not in the dict